src/base/torrentfilter.cpp

```diff
@@ -180,6 +183,10 @@ bool TorrentFilter::matchState(const BitTorrent::Torrent *const torrent) const
         return torrent->state() ==  BitTorrent::TorrentState::StalledUploading;
     case StalledDownloading:
         return torrent->state() ==  BitTorrent::TorrentState::StalledDownloading;
+    case Checking:
+        return (torrent->state() == BitTorrent::TorrentState::CheckingUploading)
+                || (torrent->state() == BitTorrent::TorrentState::CheckingDownloading)
+                || (torrent->state() == BitTorrent::TorrentState::CheckingResumeData);
```

Comment:
`torrent->isChecking()` wasn't correct, can someone confirm this implementation?

Reply:
>`torrent->isChecking()` wasn't correct

Why exactly?

Reply:
[torrent->state()](https://github.com/qbittorrent/qBittorrent/blob/9bfc74a1bcf1420ea0393ffbafe5057cae4f37e4/src/webui/api/serialize/serialize_torrent.cpp#L42-L83) is used to expose these 3 states:
- [CheckingResumeData](https://github.com/qbittorrent/qBittorrent/blob/eedd47860a7a5abb5db092774ade845a9fc1d1da/src/base/bittorrent/torrentimpl.cpp#L913-L916) 
- [CheckingUploading-CheckingResumeData](https://github.com/qbittorrent/qBittorrent/blob/eedd47860a7a5abb5db092774ade845a9fc1d1da/src/base/bittorrent/torrentimpl.cpp#L938-L944)

[isChecking()](https://github.com/qbittorrent/qBittorrent/blob/eedd47860a7a5abb5db092774ade845a9fc1d1da/src/base/bittorrent/torrentimpl.cpp#L825-L829) doesn't check the same things.

Reply:
>isChecking() doesn't check the same things.

`isChecking()` is about "hash checking" exactly.

`CheckingResumeData` is mostly an outdated concept. Currently, no "resume data checking" is actually performed. It refers to the state of the torrent initialization. I intend to rename it, but I can't get my hands on it.

Reply:
Setting aside `CheckingResumeData`, [isChecking](https://github.com/qbittorrent/qBittorrent/blob/eedd47860a7a5abb5db092774ade845a9fc1d1da/src/base/bittorrent/torrentimpl.cpp#L827) still doesn't match [this](https://github.com/qbittorrent/qBittorrent/blob/eedd47860a7a5abb5db092774ade845a9fc1d1da/src/base/bittorrent/torrentimpl.cpp#L938-L940).

Reply:
It looks like there are even more inconsistencies than I thought... I must definitely deal with it.
Well, I think you can stay with the current option for now.